hrjobcontract/js/src/job-contract/services/contract.js

```diff
@@ -281,22 +286,26 @@ define([
          * @return {Promise} resolves with the api response
          */
         fullDetails: function (contractId) {
+          var deferred = $q.defer();
+
           if (!contractId || typeof +contractId !== 'number') {
             return null;
           }
 
-          var deferred = $q.defer();
-
-          Contract.get({
-            action: 'getfulldetails',
-            json: {
-              jobcontract_id: contractId
-            }
-          }, function (data) {
-            adjustAddPublicHolidaysValue(data);
-            deferred.resolve(data);
-          }, function () {
-            deferred.reject('Could not fetch full details for contract ID:' + contractId);
+          AbsenceType.all()
+          .then(AbsenceType.loadCalculationUnits)
+          .then(function (absenceTypes) {
+            Contract.get({
+              action: 'getfulldetails',
+              json: {
+                jobcontract_id: contractId
+              }
+            }, function (contracts) {
+              adjustAddPublicHolidaysValue(contracts, _.indexBy(absenceTypes, 'id'));
+              deferred.resolve(contracts);
+            }, function () {
+              deferred.reject('Could not fetch full details for contract ID:' + contractId);
```

Comment:
Cant we use `.catch` here? instead of the 2nd callback.

Reply:
Contract.save is not a promise.